fusor-ember-cli/app/controllers/storage.js

```diff
@@ -179,6 +121,20 @@ export default Ember.Controller.extend(NeedsDeploymentMixin, {
     ]
   }),
 
+  nfsPathValidator: AllValidator.create({
+    validators: [
+      PresenceValidator.create({}),
+      NfsPathValidator.create({})
+    ]
+  }),
+
+  glusterPathValidator: AllValidator.create({
+    validators: [
+      PresenceValidator.create({}),
+      GlusterPathValidator.create({})
+    ]
+  }),
+
```

Comment:
This looks awesome. :tada: Less code and much cleaner.  If you want to rip out the duplicate code in the templates you could do something like:
```javascript
  sharePathValidator: Ember.computed('deploymentController.model.rhev_storage_type', function () {
    if (this.get('deploymentController.model.rhev_storage_type') === 'NFS') {
      return this.get('nfsPathValidator')
    }

    return this.get('glusterPathValidator');
  }),

  invalidSharePath: Ember.computed('sharePathValidator.model.rhev_share_path', 'sharePathValidator', function () {
    return !this.get('sharePathValidator').isValid(this.get('deploymentController.model.rhev_share_path'));
  }),

  invalidExportPath: Ember.computed('deploymentController.model.rhev_export_domain_path', 'sharePathValidator', function () {
    return !this.get('sharePathValidator').isValid(this.get('deploymentController.model.rhev_export_domain_path'));
  }),
```

Then you could reuse the shareValidator and use a single form.